dev/start.fish

```diff
@@ -0,0 +1,385 @@
+        else
+            powershell.exe -Command "Invoke-WebRequest -Uri '$_DOWNLOAD_URL' -OutFile '$_INSTALLER_FILE' | Out-Null"
+        end
+        if not test $status -eq 0 || not test -s "$_INSTALLER_FILE"
+            echo "Error: failed to download $_INSTALLER_DISPLAY (file missing or empty)" 1>&2
+            return 1
+        end
+    end
+
+    # installing conda
+    echo "Installing development environment..."
+    switch $_MACH
+        case Darwin Linux
+            bash "$_INSTALLER_FILE" -bfp "$_DEVENV" > /dev/null
+        case '*'
+            cmd.exe /c "start /wait \"\" \"$_INSTALLER_FILE\" /InstallationType=JustMe /RegisterPython=0 /AddToPath=0 /S /D=$_DEVENV > NUL"
+    end
+    if not test $status -eq 0
+        echo "Error: failed to install development environment" 1>&2
+        return 1
+    end
+end
+
+# create empty env if it doesn't exist
+if not test -d "$_ENV"
+    echo "Creating $_NAME..."
+    if not env PYTHONPATH="" "$_BASEEXE" create --yes --quiet "--prefix=$_ENV" > /dev/null
+        echo "Error: failed to create $_NAME" 1>&2
+        return 1
+    end
+end
+
+# check if explicitly updating or if 24 hrs since last update
+if updating
+    echo "Updating $_NAME..."
+
+    if not env PYTHONPATH="" "$_BASEEXE" update --yes --quiet --all "--prefix=$_ENV" > /dev/null
+        echo "Error: failed to update development environment" 1>&2
+        return 1
+    end
+
+    # set channels based on installer type
+    switch $_INSTALLER_TYPE
+        case miniconda
+            set _CHANNEL_NAME "defaults"
+        case miniforge
+            set _CHANNEL_NAME "conda-forge"
+    end
+
+    set install_args --yes --quiet "--prefix=$_ENV" --override-channels "--channel=$_CHANNEL_NAME" "--file=$_SRC/tests/requirements.txt" "--file=$_SRC/tests/requirements-ci.txt"
+    if test "$_MACH" = "Linux"
+        set install_args $install_args "--file=$_SRC/tests/requirements-Linux.txt"
+    end
+    set install_args $install_args "python=$_PYTHON"
+
+    if not env PYTHONPATH="" "$_BASEEXE" install $install_args > /dev/null
+        echo "Error: failed to update $_NAME" 1>&2
+        return 1
+    end
+
+    # update timestamp
+    touch "$_UPDATED"
+end
+
+# "install" conda
+# trick conda into importing from our source code and not from site-packages
+if test -z "$PYTHONPATH"
+    set -gx PYTHONPATH "$_SRC"
+else
+    set -gx PYTHONPATH "$_SRC:$PYTHONPATH"
+end
+
+# copy latest shell scripts
+echo "Update shell scripts..."
+if not "$_ENVEXE" init --install > /dev/null
+    echo "Error: failed to update shell scripts" 1>&2
+    return 1
+end
+
+# initialize conda command
+echo "Initializing shell integration..."
+eval (env CONDA_AUTO_ACTIVATE=0 "$_ENVEXE" shell.fish hook) > /dev/null
+if not test $status -eq 0
+    echo "Error: failed to initialize shell integration" 1>&2
+    return 1
+end
+
+# activate env
+echo "Activating $_NAME..."
+if not conda activate "$_ENV" > /dev/null
```

Comment:
Hmmm, I get an error here about conda not being initialized properly
```
> source ./dev/start.fish
Choose conda installer:
  1) miniconda (default - Anaconda defaults channel)
  2) miniforge (conda-forge channel)

Enter choice [1]: 2
Deactivating 1 environment(s)...
Warning: failed to deactivate environment, continuing...
Updating devenv-3.10-miniforge...

Update shell scripts...
Initializing shell integration...
Activating devenv-3.10-miniforge...

CondaError: Run 'conda init' before 'conda activate'

Error: failed to activate devenv-3.10-miniforge
```

This is happens for fish 4.x and 3.x, without an existing `devenv/`

Reply:
Thanks, this should be fixed hopefully now!